test/Microsoft.TestPlatform.Extensions.TrxLogger.UnitTests/Utility/ConverterTests.cs

```diff
@@ -116,30 +116,33 @@ public void ToTestElementShouldNotFailWhenThereIsNoTestCategoreis()
         public void ToTestElementShouldContainExpectedTestMethodPropertiesIfFqnIsSameAsTestName()
         {
             var expectedClassName = "TestProject1.Class1";
-            var fullyQualifiedName = expectedClassName + "." + "TestMethod1";
+            var expectedTestName = "TestMethod1";
+            var fullyQualifiedName = expectedClassName + "." + expectedTestName;
             var testName = "TestProject1.Class1.TestMethod1";
 
-            ValidateTestMethodProperties(testName, fullyQualifiedName, expectedClassName);
+            ValidateTestMethodProperties(testName, fullyQualifiedName, expectedClassName, expectedTestName);
         }
 
         [TestMethod]
         public void ToTestElementShouldContainExpectedTestMethodPropertiesIfFqnEndsWithTestName()
         {
             var expectedClassName = "TestProject1.Class1";
-            var fullyQualifiedName = expectedClassName + "." + "TestMethod1(2, 3, 4.0d)";
+            var expectedTestName = "TestMethod1(2, 3, 4.0d)";
+            var fullyQualifiedName = expectedClassName + "." + expectedTestName;
             var testName = "TestMethod1(2, 3, 4.0d)";
 
-            ValidateTestMethodProperties(testName, fullyQualifiedName, expectedClassName);
+            ValidateTestMethodProperties(testName, fullyQualifiedName, expectedClassName, expectedTestName);
         }
 
         [TestMethod]
         public void ToTestElementShouldContainExpectedTestMethodPropertiesIfFqnDoesNotEndsWithTestName()
         {
             var expectedClassName = "TestProject1.Class1.TestMethod1(2, 3, 4";
-            var fullyQualifiedName = "TestProject1.Class1.TestMethod1(2, 3, 4.0d)";
+            var expectedTestName = "0d)";
```

Comment:
@singhsarab I moved the test data to be more explicit, because the helper method ValidateTestMethodProperties was re- implementing the same code in the test and in the source, and both of them had the same bug. The current behavior is explicitly stating the same thing that it did before I refactored. It splits on the decimal point om `4.0d`, which I assume is not correct. What would be the correct behavior here?